app/logic/courseManagement.py

```diff
@@ -45,6 +47,56 @@ def approvedCourses(termId):
 
     return approvedCourses
 
+
+def importedCourses(termId):
+    '''
+    Queries the database to get all the necessary information for 
+    imported courses.
+    '''
+    importedCourses = (Course.select(Course, Term, CourseStatus, fn.GROUP_CONCAT(" " ,User.firstName, " ", User.lastName).alias('instructors'))
+                        .join(CourseInstructor, JOIN.LEFT_OUTER)
+                        .join(User, JOIN.LEFT_OUTER).switch(Course)
+                        .join(CourseStatus).switch(Course)
+                        .join(Term)
+                        .where(Term.id == termId, Course.status == CourseStatus.IMPORTED)
+                        .group_by(Course, Term, CourseStatus))
+
+    return importedCourses
+
+
+def editImportedCourses(courseData, attachments=None):
```

Comment:
Can we try and add comments to this function. Some of the code in here isn't necessarily intuitive.

Reply:
Fixed